kinto_megaphone/listeners.py

```diff
@@ -0,0 +1,50 @@
+from kinto.core.events import ResourceChanged
+from kinto.core.listeners import ListenerBase
+from kinto.core import utils
+from . import megaphone
+
+DEFAULT_SETTINGS = {}
+
+
+class CollectionTimestamp(ListenerBase):
+    """An event listener that pushes all collection timestamps to Megaphone."""
+    def __init__(self, client, broadcaster_id):
+        self.client = client
+        self.broadcaster_id = broadcaster_id
+
+    def __call__(self, event):
+        if not isinstance(event, ResourceChanged):
+            return
+
+        if event.payload['resource_name'] != 'record':
+            return
+
+        bucket_id = event.payload['bucket_id']
+        collection_id = event.payload['collection_id']
+        storage = event.request.registry.storage
+        parent_id = utils.instance_uri(event.request, 'collection',
+                                       bucket_id=bucket_id,
+                                       id=collection_id)
+        etag = storage.collection_timestamp('record', parent_id)
+        self.client.send_version(self.broadcaster_id,
+                                 '{}_{}'.format(bucket_id, collection_id),
```

Comment:
we should use another separator, like `.`  since underscores are valid characters for bucket names

Reply:
A Megaphone channel ID has to conform to `URLSAFE_BASE64`, which is `^[A-Za-z0-9\-_]+$`. So there's no separator we can use that isn't also a valid bucket name.